sympy/angles/euler.py

```diff
@@ -0,0 +1,688 @@
+import warnings
+from sympy.utilities.iterables import iterable
+from sympy.matrices.dense import MutableDenseMatrix as Matrix
+from sympy.core.numbers import pi
+from sympy.algebras.quaternion import Quaternion
+from sympy.core.sympify import sympify, _sympify
+from sympy.core.expr import Expr
+from sympy.core.symbol import Symbol
+from sympy.simplify.trigsimp import trigsimp
+from sympy.matrices.dense import rot_axis1, rot_axis2, rot_axis3
+
+from mpmath.libmp.libmpf import prec_to_dps
+
+
+def _check_sequence(seq):
+    """validate seq and return info"""
+    if isinstance(seq, Symbol):
+        seq = str(seq)
+    elif type(seq) != str:
+        raise ValueError('Expected seq to be a string.')
+    if len(seq) != 3:
+        raise ValueError("Expected 3 axes, got `{}`.".format(seq))
+
+    intrinsic = seq.isupper()
+    extrinsic = seq.islower()
+    if not (intrinsic or extrinsic):
+        raise ValueError("seq must either be fully uppercase (for extrinsic "
+                         "rotations), or fully lowercase, for intrinsic "
+                         "rotations).")
+
+    i, j, k = seq.lower()
+    if i == j or j == k:
+        raise ValueError("Consecutive axes must be different")
+
+    bad = set(seq) - set('xyzXYZ')
+    if bad:
+        raise ValueError("Expected axes from `seq` to be from "
+                         "['x', 'y', 'z'] or ['X', 'Y', 'Z'], "
+                         "got {}".format(''.join(bad)))
+
+
+def _wrap(angle):
+    if angle.is_number:
+        return (angle + pi) % (2 * pi) - pi
+    else:
+        return angle
+
+
+class Euler(Expr):
```

Comment:
This should not subclass Expr. Expr means things that can be added, multiplied etc and can be in the args of an Add or Mul. This class does not make sense in any of those contexts.

Reply:
Alright, what should I use as a mother class?

Reply:
It should probably be Basic.